test/address_test.rb

```diff
@@ -125,6 +125,17 @@ def initialize
 
       expect(Yast::Host.names(new_ip)).to eql original_names
     end
+
+    it "do not crash when ip was changed, no hostnames exist and the hostname is empty" do
```

Comment:
does

Reply:
It is true that it does not crash now, but I would use contexts explaining the different scenarios and what does each one:

```ruby

context "when the ifcfg ipaddr has changed" do
  context "but the associated hostname not" do
   ...
  end
  context "and the hostname as well" do
  end
end
context "when neither the ifcpfg ipaddr nor associated hostnames have changed" do
  it "does not touch the hosts file"
    expect(Host).to not_receive(:names)
  end
end
```
...